internal/ingest/upload.go

```diff
@@ -117,3 +137,61 @@ func (w *goaWrapper) initSIP(
 
 	return nil
 }
+
+func (w *goaWrapper) findOrCreateUser(
+	ctx context.Context,
+	perSvc persistence.Service,
+) (*datatypes.User, error) {
+	claims := auth.UserClaimsFromContext(ctx)
+	if claims == nil {
+		return nil, nil
+	}
+	if claims.ISS == "" {
+		return nil, fmt.Errorf("invalid user claims: missing ISS")
+	}
+	if claims.Sub == "" {
+		return nil, fmt.Errorf("invalid user claims: missing Sub")
+	}
+
+	user, err := perSvc.ReadJWTUser(ctx, claims.ISS, claims.Sub)
```

Comment:
`ReadJWTUser` seems to be like it's going to get another type of user. What about `ReadUserByClaims` and passing the full claims? Or `ReadUserByOIDC` and keeping the same signature?